xbmc/platform/linux/SysfsPath.h

```diff
@@ -0,0 +1,87 @@
+/*
+ *  Copyright (C) 2011-2018 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#pragma once
+
+#include "utils/log.h"
+
+#include <fstream>
+#include <string>
+
+class CSysfsPath
+{
+public:
+  CSysfsPath() = default;
+  CSysfsPath(const std::string& path) : m_path(path) {}
+  ~CSysfsPath() = default;
+
+  void operator=(const std::string& path) { m_path = path; }
+
+  template<typename T>
+  bool Get(T& value)
+  {
+    return GetPrivate(value);
+  }
+
+  template<typename T>
+  bool Set(T value)
+  {
+    return SetPrivate(value);
+  }
+
+private:
+  template<typename T>
+  bool GetPrivate(T& value)
```

Comment:
I'm against out reference parameters personally. It makes the code using this function very weird to read. With C++17 I'd go for `std::optional<T>` as return type but as we can't use that yet what about `std::tuple<bool, T>`? Or, alternatively a) use exceptions or b) have a separate function to check for the existence of the path.

Reply:
I agree and this is something I struggled with. I opted for the simple path check. It would be nice to use `std::optional<T>` when it's available to use.